server.go

```diff
@@ -201,7 +201,9 @@ func (server *mongoServer) Connect(timeout time.Duration, socket *mongoSocket) e
 	var socketExpiryTime *time.Time
 	if server.maxSocketReuseTime != 0 {
 		duration := server.maxSocketReuseTime
-		durationWithJitter := time.Duration(float64(duration) * (1 + rand.Float64()*socketExpiryJitterAmount))
+		randomSource := rand.NewSource(time.Now().UnixNano())
```

Comment:
Also it could be rewritten as `rand.Seed(time.Now().Unix())` https://golang.org/pkg/math/rand/#Seed which is threadsafe.

Reply:
yeah I think rand.Seed seems better, but we only need to call it once during initialization. And rest of the code should remain unchanged as getting random number is already protected (https://golang.org/src/math/rand/rand.go?s=7744:7764#L317)